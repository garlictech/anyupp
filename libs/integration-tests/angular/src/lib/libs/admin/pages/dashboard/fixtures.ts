import { ProductComponentSetType, ProductType } from '@bgap/domain';

export const printableOrder = {
  id: 'seeded_order_id_6',
  createdAt: '2022-01-31T15:04:37.521Z',
  userId: 'seeded_user_id_6',
  unitId: 'seeded_unit_c1_g1_1_id',
  orderNum: '000001',
  items: [
    {
      productId: 'seeded_chain_product_c1_1_id',
      variantId: 'seeded_chain_product_variant_c1_p1_1_id',
      created: null,
      productName: {
        en: 'Hamburger',
        de: 'Hamburger',
        hu: 'Hamburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['egg', 'gluten', 'mustard', 'milk', 'soya', 'fish', 'sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_2_id',
      variantId: '16853de0-528d-11ec-90c0-fdfe82b3802d',
      created: null,
      productName: {
        en: 'Fishburger',
        de: 'Fishburger',
        hu: 'Halburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: [
        'lupin',
        'molluscs',
        'crustaceans',
        'egg',
        'soya',
        'sesame',
        'fish',
        'gluten',
      ],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
    {
      productId: 'seeded_chain_product_c1_3_id',
      variantId: 'd22dee40-528b-11ec-afe3-15e1afd659e2',
      created: null,
      productName: {
        en: 'Sajtburger',
        de: 'Sajtburger',
        hu: 'Sajtburger',
      },
      image: null,
      quantity: 5,
      statusLog: [
        {
          userId: 'test-alice',
          status: 'none',
          ts: 1627909024677,
        },
        {
          userId: 'test-alice',
          status: 'placed',
          ts: 1627909024677,
        },
      ],
      variantName: {
        en: 'normal',
        de: 'normal',
        hu: 'normál',
      },
      laneId: 'lane_01',
      allergens: ['sesame'],
      configSets: [
        {
          productSetId: 'test_product_component_set_id',
          name: {
            en: 'Modifier comp set',
            de: null,
            hu: 'Módosító komponens set',
          },
          type: ProductComponentSetType.modifier,
          items: [
            {
              productComponentId: 'test_product_component_id',
              price: -1.7999999999999998,
              name: {
                en: 'Room temperature',
                de: 'Room temperature',
                hu: 'Szobahőmérsékletű',
              },
              allergens: ['egg', 'gluten'],
              netPackagingFee: null,
            },
          ],
        },
      ],
      productType: ProductType.drink,
      externalId: null,
      netPackagingFee: null,
      serviceFee: null,
      priceShown: {
        currency: 'HUF',
        pricePerUnit: 300,
        priceSum: 1500,
        tax: 27,
        taxSum: 318.9,
      },
      sumPriceShown: {
        currency: 'HUF',
        pricePerUnit: 298.2,
        priceSum: 1491,
        tax: 27,
        taxSum: 316.98,
      },
    },
  ],
  paymentMode: {
    type: 'cash',
    caption: null,
    method: 'cash',
  },
  statusLog: [
    {
      userId: 'test-alice',
      status: 'placed',
      ts: 1627909024677,
    },
  ],
  archived: false,
  takeAway: false,
  place: {
    seat: '02',
    table: '01',
  },
  paymentIntention: null,
  transactionStatus: 'success',
  transactionId: 'seeded_transaction_id_6',
  unpayCategory: null,
  orderMode: 'instant',
  servingMode: 'inPlace',
  serviceFee: {
    currency: 'HUF',
    netPrice: 200,
    taxPercentage: 27,
  },
  packagingSum: null,
  sumPriceShown: {
    currency: 'HUF',
    pricePerUnit: 298.2,
    priceSum: 2982,
    tax: 27,
    taxSum: 633.96,
  },
  rating: {
    key: 'RATING KEY',
    value: 3,
  },
  hasRated: true,
  tip: null,
  tipTransactionStatus: null,
  tipTransactionId: null,
  updatedAt: '2022-01-31T15:04:37.521Z',
  transaction: {
    id: 'seeded_transaction_id_6',
    userId: 'test-alice',
    orderId: 'seeded_order_id_6',
    type: 'cash',
    total: 0,
    currency: 'HUF',
    status: 'waiting_for_payment',
    externalTransactionId: null,
    invoiceId: null,
    receiptId: null,
    createdAt: '2022-01-31T15:04:37.267Z',
    updatedAt: '2022-01-31T15:04:37.267Z',
    paymentMethodId: null,
    user: {
      id: 'test-alice',
      name: 'Mekk Elek',
      email: 'testuser+alice@anyupp.com',
      phone: '1234',
      profileImage: null,
      stripeCustomerId: null,
      invoiceAddress: null,
      createdAt: '2022-01-31T15:04:19.739Z',
      updatedAt: '2022-01-31T15:04:19.739Z',
    },
    invoice: null,
    receipt: null,
  },
  tipTransaction: null,
};
