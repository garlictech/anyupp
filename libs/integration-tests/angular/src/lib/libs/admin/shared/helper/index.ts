import { Auth } from 'aws-amplify';

import { awsConfig } from '@bgap/crud-gql/api';
import {
  getCognitoUsername,
  testAdminUserPassword,
} from '@bgap/shared/fixtures';

export const signInToCognito = async () => {
  Auth.configure({
    ...awsConfig,
    authenticationFlowType: 'USER_PASSWORD_AUTH',
    aws_appsync_authenticationType: 'AMAZON_COGNITO_USER_POOLS',
  });

  await Auth.signIn(getCognitoUsername('monad'), testAdminUserPassword);
};

export const signOutFromCognito = async () => {
  await Auth.signOut();
};
