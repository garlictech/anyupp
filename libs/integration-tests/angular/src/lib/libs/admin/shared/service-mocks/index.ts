export class MockNbDialogService {
  open() {
    /** */
  }
}

export class MockTranslateService {
  get() {
    /** */
  }
}
