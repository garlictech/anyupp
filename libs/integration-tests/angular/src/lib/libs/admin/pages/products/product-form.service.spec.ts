import { combineLatest, EMPTY, of } from 'rxjs';
import { catchError, switchMap, tap } from 'rxjs/operators';

import { TestBed } from '@angular/core/testing';
import { FormArray, ReactiveFormsModule } from '@angular/forms';
import { ProductFormService } from '@bgap/admin/pages/products';
import { CrudSdkService } from '@bgap/admin/shared/data-access/sdk';
import * as CrudApi from '@bgap/crud-gql/api';
import { productFixture, testIdPrefix } from '@bgap/shared/fixtures';
import { EProductLevel, UpsertResponse } from '@bgap/shared/types';
import { StoreModule } from '@ngrx/store';

import { signInToCognito, signOutFromCognito } from '../../shared/helper';

describe('ProductFormService', () => {
  const chainProductId = `${testIdPrefix}ADMIN_PRODUCT_IT_CHAIN_PRODUCT_ID_01`;
  const groupProductId = `${testIdPrefix}ADMIN_PRODUCT_IT_GROUP_PRODUCT_ID_01`;
  const unitProductId = `${testIdPrefix}ADMIN_PRODUCT_IT_UNIT_PRODUCT_ID_01`;

  let service: ProductFormService;
  let crudSdk: CrudSdkService;

  const cleanup = () =>
    combineLatest([
      crudSdk.sdk.DeleteChainProduct({
        input: { id: chainProductId },
      }),
      crudSdk.sdk.DeleteGroupProduct({
        input: { id: groupProductId },
      }),
      crudSdk.sdk.DeleteUnitProduct({
        input: { id: unitProductId },
      }),
    ]);

  beforeAll(async () => {
    await signInToCognito();
  });

  afterAll(async () => {
    await signOutFromCognito();
  });

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [ReactiveFormsModule, StoreModule.forRoot({})],
    });

    service = TestBed.inject(ProductFormService);
    crudSdk = TestBed.inject(CrudSdkService);
  });

  it('createProductFormGroup should create form group', () => {
    expect(service.createProductFormGroup().value).toMatchSnapshot();
  });

  it('createProductExtendFormGroup should create form group', () => {
    expect(
      service.createProductExtendFormGroup(EProductLevel.GROUP).value,
    ).toMatchSnapshot();
  });

  it('createProductExtendFormGroup should create form group', () => {
    expect(
      service.createProductExtendFormGroup(EProductLevel.UNIT).value,
    ).toMatchSnapshot();
  });

  it('patchProductVariants should add variants to form array', () => {
    const variantsArray = new FormArray([]);

    service.patchProductVariants(
      productFixture.chainProductBase.variants,
      variantsArray,
    );

    expect(variantsArray.value).toMatchSnapshot();
  });

  it('patchExtendedProductVariants should add variants to form array', () => {
    const variantsArray = new FormArray([]);

    service.patchExtendedProductVariants(
      productFixture.groupProductBase.variants,
      variantsArray,
    );

    expect(variantsArray.value).toMatchSnapshot();
  });

  it('patchConfigSet should add configsets to form array', () => {
    const configSetsArray = new FormArray([]);

    service.patchConfigSet([], configSetsArray);

    expect(configSetsArray.value).toMatchSnapshot();
  });

  it('saveChainForm$ should call createChainProduct$ method when id is not specified', done => {
    const createSpy = jest
      .spyOn(service, 'createChainProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service.saveChainForm$(productFixture.chainProductInputBase).subscribe();

    expect(createSpy).toHaveBeenCalledWith({
      ...productFixture.chainProductInputBase,
    });

    done();
  });

  it('saveChainForm$ should call updateChainProduct$ method when id is specified', done => {
    const updateSpy = jest
      .spyOn(service, 'updateChainProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveChainForm$(
        productFixture.chainProductInputBase,
        chainProductId,
        false,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productFixture.chainProductInputBase,
      id: chainProductId,
      dirty: false,
    });

    done();
  });

  it('saveChainForm$ should call updateChainProduct$ method with avoided dirty flag', done => {
    const updateSpy = jest
      .spyOn(service, 'updateChainProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveChainForm$(productFixture.chainProductInputBase, chainProductId)
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith({
      ...productFixture.chainProductInputBase,
      id: chainProductId,
      dirty: undefined,
    });

    done();
  });

  it('createChainProduct$ should create chain product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createChainProduct$({
            ...productFixture.chainProductInputBase,
            id: chainProductId,
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);

  it('updateChainProduct$ should update chain product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createChainProduct$({
            ...productFixture.chainProductInputBase,
            id: chainProductId,
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.ChainProduct>>saveResponse).data.id
            ? service.updateChainProduct$({
                ...productFixture.chainProductInputBase,
                id: chainProductId,
                image: 'IMAGE_MOD',
                productType: 'PRODUCT_TYPE_MOD',
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);

  it('saveGroupExtendForm$ should call createGroupProduct$ method when editing mode is false', done => {
    const createSpy = jest
      .spyOn(service, 'createGroupProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service
      .saveGroupExtendForm$(
        productFixture.groupProductInputBase,
        productFixture.groupProductInputBase,
        false,
      )
      .subscribe();

    expect(createSpy).toHaveBeenCalledWith(
      productFixture.groupProductInputBase,
    );

    done();
  });

  it('saveGroupExtendForm$ should call updateGroupProduct$ method when editing mode is true', done => {
    const updateSpy = jest
      .spyOn(service, 'updateGroupProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveGroupExtendForm$(
        productFixture.groupProductInputBase,
        productFixture.groupProductInputBase,
        true,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith(
      productFixture.groupProductInputBase,
    );

    done();
  });

  it('createGroupProduct$ should create group product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createGroupProduct$({
            ...productFixture.groupProductInputBase,
            id: groupProductId,
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);

  it('updateGroupProduct$ should update group product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createGroupProduct$({
            ...productFixture.groupProductInputBase,
            id: groupProductId,
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.GroupProduct>>saveResponse).data.id
            ? service.updateGroupProduct$({
                ...productFixture.groupProductInputBase,
                id: groupProductId,
                tax: 33,
                takeawayTax: 30,
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);

  it('saveUnitExtendForm$ should call createUnitProduct$ method when editing mode is false', done => {
    const createSpy = jest
      .spyOn(service, 'createUnitProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'insert' }));

    service
      .saveUnitExtendForm$(
        productFixture.unitProductInputBase,
        productFixture.unitProductInputBase,
        false,
      )
      .subscribe();

    expect(createSpy).toHaveBeenCalledWith(productFixture.unitProductInputBase);

    done();
  });

  it('saveUnitExtendForm$ should call updateUnitProduct$ method when editing mode is true', done => {
    const updateSpy = jest
      .spyOn(service, 'updateUnitProduct$')
      .mockImplementationOnce(() => of({ data: 'ok', type: 'update' }));

    service
      .saveUnitExtendForm$(
        productFixture.unitProductInputBase,
        productFixture.unitProductInputBase,
        true,
      )
      .subscribe();

    expect(updateSpy).toHaveBeenCalledWith(productFixture.unitProductInputBase);

    done();
  });

  it('createUnitProduct$ should create unit product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createUnitProduct$({
            ...productFixture.unitProductInputBase,
            id: unitProductId,
          }),
        ),
        tap(saveResponse => {
          expect(saveResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);

  it('updateUnitProduct$ should update unit product', done => {
    cleanup()
      .pipe(
        switchMap(() =>
          service.createUnitProduct$({
            ...productFixture.unitProductInputBase,
            id: unitProductId,
          }),
        ),
        catchError(() => cleanup()),
        switchMap(saveResponse =>
          (<UpsertResponse<CrudApi.UnitProduct>>saveResponse).data.id
            ? service.updateUnitProduct$({
                ...productFixture.unitProductInputBase,
                id: unitProductId,
                supportedServingModes: [CrudApi.ServingMode.inplace],
                isVisible: false,
              })
            : EMPTY,
        ),
        catchError(() => cleanup()),
        tap(updateResponse => {
          expect(updateResponse).toMatchSnapshot({
            data: {
              createdAt: expect.any(String),
              updatedAt: expect.any(String),
            },
          });
        }),
        switchMap(() => cleanup()),
      )
      .subscribe(() => {
        done();
      });
  }, 25000);
});
