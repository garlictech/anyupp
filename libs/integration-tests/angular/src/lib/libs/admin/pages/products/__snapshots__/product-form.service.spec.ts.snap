// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductFormService createChainProduct$ should create chain product 1`] = `
Object {
  "data": Object {
    "allergens": Array [
      "egg",
      "gluten",
    ],
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "description": Object {
      "de": null,
      "en": "DESCRIPTION",
      "hu": null,
    },
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_IT_CHAIN_PRODUCT_ID_01",
    "image": "IMAGE",
    "isVisible": true,
    "name": Object {
      "de": null,
      "en": "CHAIN_PRODUCT",
      "hu": null,
    },
    "productCategoryId": "productCategoryId_",
    "productType": "PRODUCT_TYPE",
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 0.5,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_ChainProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 1,
          "unit": "CHAIN_UNIT",
        },
        "position": 1,
        "price": 10,
        "refGroupPrice": 10,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_ChainProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 2,
          "unit": "CHAIN_UNIT",
        },
        "position": 2,
        "price": 20,
        "refGroupPrice": 20,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "insert",
}
`;

exports[`ProductFormService createGroupProduct$ should create group product 1`] = `
Object {
  "data": Object {
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "dirty": null,
    "externalId": null,
    "groupId": "groupId_",
    "id": "test_ADMIN_PRODUCT_IT_GROUP_PRODUCT_ID_01",
    "isVisible": true,
    "parentId": "parentId_",
    "takeawayTax": 12,
    "tax": 27,
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_GroupProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 1,
          "unit": "GROUP_UNIT",
        },
        "position": 1,
        "price": 20,
        "refGroupPrice": 20,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 2,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_GroupProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 2,
          "unit": "GROUP_UNIT",
        },
        "position": 2,
        "price": 40,
        "refGroupPrice": 40,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "insert",
}
`;

exports[`ProductFormService createProductExtendFormGroup should create form group 1`] = `
Object {
  "configSets": Array [],
  "isVisible": "",
  "takeawayTax": "",
  "tax": "",
  "variants": Array [],
}
`;

exports[`ProductFormService createProductExtendFormGroup should create form group 2`] = `
Object {
  "configSets": Array [],
  "isVisible": "",
  "laneId": "",
  "supportedServingModes": Array [],
  "variants": Array [],
}
`;

exports[`ProductFormService createProductFormGroup should create form group 1`] = `
Object {
  "allergens": Array [],
  "configSets": Array [],
  "description": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "image": "",
  "isVisible": "",
  "name": Object {
    "de": "",
    "en": "",
    "hu": "",
  },
  "productCategoryId": "",
  "productType": "",
  "variants": Array [],
}
`;

exports[`ProductFormService createUnitProduct$ should create unit product 1`] = `
Object {
  "data": Object {
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "dirty": null,
    "externalId": null,
    "groupId": "groupId_",
    "id": "test_ADMIN_PRODUCT_IT_UNIT_PRODUCT_ID_01",
    "isVisible": true,
    "laneId": "laneId_",
    "parentId": "parentId_",
    "position": 1,
    "supportedServingModes": Array [
      "takeAway",
    ],
    "takeaway": true,
    "unitId": "unitId_",
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1.5,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_UnitProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": 1,
        "pack": Object {
          "size": 1,
          "unit": "UNIT_UNIT",
        },
        "position": 1,
        "price": 30,
        "refGroupPrice": 30,
        "soldOut": true,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 3,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_UnitProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": 1,
        "pack": Object {
          "size": 2,
          "unit": "UNIT_UNIT",
        },
        "position": 2,
        "price": 60,
        "refGroupPrice": 60,
        "soldOut": true,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "insert",
}
`;

exports[`ProductFormService patchConfigSet should add configsets to form array 1`] = `Array []`;

exports[`ProductFormService patchExtendedProductVariants should add variants to form array 1`] = `
Array [
  Object {
    "availabilities": Array [
      Object {
        "dayFrom": "",
        "dayTo": "",
        "price": 1,
        "timeFrom": "",
        "timeTo": "",
        "type": "A",
      },
    ],
    "id": "test_GroupProductVariant_id_1",
    "isAvailable": true,
    "netPackagingFee": 0,
    "pack": Object {
      "size": 1,
      "unit": "GROUP_UNIT",
    },
    "position": 1,
    "price": 20,
    "refGroupPrice": 20,
    "variantName": Object {
      "de": "",
      "en": "VARIANT_NAME_1",
      "hu": "",
    },
  },
  Object {
    "availabilities": Array [
      Object {
        "dayFrom": "",
        "dayTo": "",
        "price": 2,
        "timeFrom": "",
        "timeTo": "",
        "type": "A",
      },
    ],
    "id": "test_GroupProductVariant_id_2",
    "isAvailable": true,
    "netPackagingFee": 0,
    "pack": Object {
      "size": 2,
      "unit": "GROUP_UNIT",
    },
    "position": 2,
    "price": 40,
    "refGroupPrice": 40,
    "variantName": Object {
      "de": "",
      "en": "VARIANT_NAME_2",
      "hu": "",
    },
  },
]
`;

exports[`ProductFormService patchProductVariants should add variants to form array 1`] = `
Array [
  Object {
    "availabilities": Array [],
    "id": "test_ChainProductVariant_id_1",
    "isAvailable": true,
    "netPackagingFee": 0,
    "pack": Object {
      "size": 1,
      "unit": "CHAIN_UNIT",
    },
    "position": 1,
    "price": 10,
    "refGroupPrice": 10,
    "variantName": Object {
      "de": "",
      "en": "VARIANT_NAME_1",
      "hu": "",
    },
  },
  Object {
    "availabilities": Array [],
    "id": "test_ChainProductVariant_id_2",
    "isAvailable": true,
    "netPackagingFee": 0,
    "pack": Object {
      "size": 2,
      "unit": "CHAIN_UNIT",
    },
    "position": 2,
    "price": 20,
    "refGroupPrice": 20,
    "variantName": Object {
      "de": "",
      "en": "VARIANT_NAME_2",
      "hu": "",
    },
  },
]
`;

exports[`ProductFormService updateChainProduct$ should update chain product 1`] = `
Object {
  "data": Object {
    "allergens": Array [
      "egg",
      "gluten",
    ],
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "description": Object {
      "de": null,
      "en": "DESCRIPTION",
      "hu": null,
    },
    "dirty": null,
    "externalId": null,
    "id": "test_ADMIN_PRODUCT_IT_CHAIN_PRODUCT_ID_01",
    "image": "IMAGE_MOD",
    "isVisible": true,
    "name": Object {
      "de": null,
      "en": "CHAIN_PRODUCT",
      "hu": null,
    },
    "productCategoryId": "productCategoryId_",
    "productType": "PRODUCT_TYPE_MOD",
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 0.5,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_ChainProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 1,
          "unit": "CHAIN_UNIT",
        },
        "position": 1,
        "price": 10,
        "refGroupPrice": 10,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_ChainProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 2,
          "unit": "CHAIN_UNIT",
        },
        "position": 2,
        "price": 20,
        "refGroupPrice": 20,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "update",
}
`;

exports[`ProductFormService updateGroupProduct$ should update group product 1`] = `
Object {
  "data": Object {
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "dirty": null,
    "externalId": null,
    "groupId": "groupId_",
    "id": "test_ADMIN_PRODUCT_IT_GROUP_PRODUCT_ID_01",
    "isVisible": true,
    "parentId": "parentId_",
    "takeawayTax": 30,
    "tax": 33,
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_GroupProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 1,
          "unit": "GROUP_UNIT",
        },
        "position": 1,
        "price": 20,
        "refGroupPrice": 20,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 2,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_GroupProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": null,
        "pack": Object {
          "size": 2,
          "unit": "GROUP_UNIT",
        },
        "position": 2,
        "price": 40,
        "refGroupPrice": 40,
        "soldOut": null,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "update",
}
`;

exports[`ProductFormService updateUnitProduct$ should update unit product 1`] = `
Object {
  "data": Object {
    "chainId": "chainId_",
    "configSets": null,
    "createdAt": Any<String>,
    "dirty": null,
    "externalId": null,
    "groupId": "groupId_",
    "id": "test_ADMIN_PRODUCT_IT_UNIT_PRODUCT_ID_01",
    "isVisible": false,
    "laneId": "laneId_",
    "parentId": "parentId_",
    "position": 1,
    "supportedServingModes": Array [
      "inPlace",
    ],
    "takeaway": true,
    "unitId": "unitId_",
    "updatedAt": Any<String>,
    "variants": Array [
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 1.5,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_UnitProductVariant_id_1",
        "isAvailable": true,
        "netPackagingFee": 1,
        "pack": Object {
          "size": 1,
          "unit": "UNIT_UNIT",
        },
        "position": 1,
        "price": 30,
        "refGroupPrice": 30,
        "soldOut": true,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_1",
          "hu": null,
        },
      },
      Object {
        "availabilities": Array [
          Object {
            "dayFrom": "",
            "dayTo": "",
            "price": 3,
            "timeFrom": "",
            "timeTo": "",
            "type": "A",
          },
        ],
        "id": "test_UnitProductVariant_id_2",
        "isAvailable": true,
        "netPackagingFee": 1,
        "pack": Object {
          "size": 2,
          "unit": "UNIT_UNIT",
        },
        "position": 2,
        "price": 60,
        "refGroupPrice": 60,
        "soldOut": true,
        "variantName": Object {
          "de": null,
          "en": "VARIANT_NAME_2",
          "hu": null,
        },
      },
    ],
  },
  "type": "update",
}
`;
